lib/gems/pending/VMwareWebService/VimService.rb

```diff
@@ -1249,16 +1253,32 @@ def marshalObj(node, obj)
     end
   end
 
+  def handle_memory_and_gc(response)
+    # At this point we don't need the internal raw XML content since we
+    #   have the parsed XML document, so we can free it for the GC.
+    response.instance_variable_set(:@http_body, nil)
+
+    xml_len = response.document.to_s.length
```

Comment:
Perhaps instead of .to_s on the document, we can use the `@http_body` directly...

```ruby
  def handle_memory_and_gc(response)
    xml_len = response.instance_variable_get(:@http_body).length

    # At this point we don't need the internal raw XML content since we
    #   have the parsed XML document, so we can free it for the GC.
    response.instance_variable_set(:@http_body, nil)
```